src/main/java/net/sf/jabref/gui/journals/ManageJournalAbbreviationsView.java

```diff
@@ -0,0 +1,543 @@
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+package net.sf.jabref.gui.journals;
+
+import java.io.File;
+
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.beans.value.ChangeListener;
+import javafx.beans.value.ObservableValue;
+import javafx.event.EventHandler;
+import javafx.fxml.FXML;
+import javafx.scene.control.Alert.AlertType;
+import javafx.scene.control.Button;
+import javafx.scene.control.ButtonBar;
+import javafx.scene.control.ButtonBar.ButtonData;
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.DialogPane;
+import javafx.scene.control.Label;
+import javafx.scene.control.ProgressIndicator;
+import javafx.scene.control.TableCell;
+import javafx.scene.control.TableColumn;
+import javafx.scene.control.TableView;
+import javafx.scene.control.TextField;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.KeyEvent;
+import javafx.scene.text.Text;
+import javafx.stage.FileChooser;
+import javafx.stage.FileChooser.ExtensionFilter;
+import javafx.stage.Stage;
+
+import net.sf.jabref.JabRefException;
+import net.sf.jabref.gui.FXAlert;
+import net.sf.jabref.gui.FXDialogs;
+import net.sf.jabref.gui.IconTheme;
+import net.sf.jabref.logic.l10n.Localization;
+
+import com.airhacks.afterburner.views.FXMLView;
+
+/**
+ * This class controls the user interface of the journal abbreviations dialog.
+ * The ui elements and their layout are defined in the fxml file in the resource folder.
+ */
+public class ManageJournalAbbreviationsView extends FXMLView {
+
+    private final ManageJournalAbbreviationsViewModel viewModel = new ManageJournalAbbreviationsViewModel();
+    private final BooleanProperty isEditableAndRemovable = new SimpleBooleanProperty(true);
+
+    @FXML
+    private TableView<AbbreviationViewModel> journalAbbreviationsTable;
+    @FXML
+    private TableColumn<AbbreviationViewModel, String> journalTableNameColumn;
+    @FXML
+    private TableColumn<AbbreviationViewModel, String> journalTableAbbreviationColumn;
+    @FXML
+    private TableColumn<AbbreviationViewModel, Boolean> journalTableEditColumn;
+    @FXML
+    private TableColumn<AbbreviationViewModel, Boolean> journalTableDeleteColumn;
+    @FXML
+    private Button cancelButton;
+    @FXML
+    private Button saveButton;
+    @FXML
+    private ComboBox<AbbreviationsFile> journalFilesBox;
+    @FXML
+    private Button addJournalFileButton;
+    @FXML
+    private Button addNewJournalFileButton;
+    @FXML
+    private Button removeJournalAbbreviationsButton;
+    @FXML
+    private ProgressIndicator progressIndicator;
+    @FXML
+    private Label loadingLabel;
+
+
+    public ManageJournalAbbreviationsView() {
+        super();
+        bundle = Localization.getMessages();
+    }
+
+    @FXML
+    private void initialize() {
+        setUpTable();
+        setBindings();
+        setButtonStyles();
+        new Thread(new Runnable() {
```

Comment:
JavaFX seems to have a new interface for background tasks: docs.oracle.com/javafx/2/threads/jfxpub-threads.htm
To be honest, I have no idea about its advantages/disadvantages. Could you pls do some research and maybe switch the code to `Task<...>`